scripts/setup-azure.sh

```diff
@@ -23,44 +23,92 @@ setup_service_principal() {
     local subscription_name=$(az account show --query name -o tsv)
     
     print_status "Current subscription: $subscription_name ($subscription_id)"
-    
-    read -p "Do you want to create a new service principal? (y/n): " -n 1 -r
-    echo
-    if [[ $REPLY =~ ^[Yy]$ ]]; then
-        local sp_name="ButtercupCRS-$(date +%Y%m%d-%H%M%S)"
-        print_status "Creating service principal: $sp_name"
-        
-        local sp_output=$(az ad sp create-for-rbac --name "$sp_name" --role Contributor --scopes "/subscriptions/$subscription_id" --output json)
-        
-        local app_id=$(echo "$sp_output" | jq -r '.appId')
-        local password=$(echo "$sp_output" | jq -r '.password')
-        local tenant_id=$(echo "$sp_output" | jq -r '.tenant')
-        
-        print_success "Service principal created successfully"
-        echo
-        print_status "Service Principal Details:"
-        echo "  Name: $sp_name"
-        echo "  App ID: $app_id"
-        echo "  Tenant ID: $tenant_id"
-        echo "  Password: $password"
+
+    # Source deployment/env to get existing RESOURCE_GROUP_NAME if set
+    if [ -f "deployment/env" ]; then
```

Comment:
Should we output a warning and ask the user if they want to create one before starting to do the individual steps?

Reply:
Might be better, but this function is called only after `setup_config`, thus the config file should be there already.